api/scpca_portal/test/management/commands/test_load_metadata.py

```diff
@@ -0,0 +1,297 @@
+
+        assert_object_count(objects_loaded=False)
+
+        # Make sure reloading works smoothly.
+        self.load_metadata(
+            clean_up_input_data=False,
+            reload_all=False,
+            reload_existing=True,
+            scpca_project_id=project_id,
+            update_s3=False,
+        )
+        assert_object_count()
+
+    def test_single_cell_metadata(self):
+        project_id = "SCPCP999990"
+        self.load_metadata(
+            clean_up_input_data=False,
+            reload_all=False,
+            reload_existing=False,
+            scpca_project_id=project_id,
+            update_s3=False,
+        )
+
+        project = Project.objects.get(scpca_id=project_id)
+        self.assertProjectData(project)
+        self.assertFalse(project.has_cite_seq_data)
+        self.assertTrue(project.includes_anndata)
+        self.assertTrue(project.modalities)
+        self.assertEqual(project.multiplexed_sample_count, 0)
+        self.assertEqual(project.organisms, ["Homo sapiens"])
+        # This project contains 3 samples
+        single_cell = 2
+        spatial = 1
+        bulk = 1
+        expected_samples = single_cell + spatial + bulk
+        self.assertEqual(project.sample_count, expected_samples)
+        self.assertFalse(project.has_multiplexed_data)
+        self.assertEqual(project.sample_count, 4)
+        self.assertEqual(project.seq_units, "cell, spot")
+        self.assertEqual(project.summaries.count(), 4)
+        self.assertEqual(project.summaries.first().sample_count, 1)
+        self.assertEqual(project.technologies, "10Xv3, visium")
+
+        sample = project.samples.filter(has_single_cell_data=True).first()
+        self.assertIsNone(sample.demux_cell_count_estimate)
+        self.assertFalse(sample.has_bulk_rna_seq)
+        self.assertFalse(sample.has_cite_seq_data)
+        # This line will probably fail when switching test data versions
+        # The reason is that the filtered_cells attribute from the library json files,
+        # from which sample_cell_count_estimate is calculated, changes from version to version
+        self.assertEqual(sample.sample_cell_count_estimate, 3432)
+        self.assertEqual(sample.seq_units, "cell")
+        self.assertEqual(sample.technologies, "10Xv3")
+
+        expected_additional_metadata_keys = [
+            "development_stage_ontology_term_id",
+            "disease_ontology_term_id",
+            "organism",
+            "organism_ontology_id",
+            "participant_id",
+            "self_reported_ethnicity_ontology_term_id",
+            "sex_ontology_term_id",
+            "submitter_id",
+            "tissue_ontology_term_id",
+            "WHO_grade",
+        ]
+        self.assertEqual(
+            expected_additional_metadata_keys, project.additional_metadata_keys.split(", ")
+        )
+        self.assertEqual(
+            set(expected_additional_metadata_keys), set(sample.additional_metadata.keys())
+        )
```

Comment:
Based on previous conversations, I feel like we'd want to delete these metadata key assertion tests (also in `test_spatial_metadata` and `test_multiplexed_metadata`). At some point soon we're going to invert this logic in any case, but in general we discussed not explicitly enumerating keys that we're not explicitly saving.